package clientmsg;
import "common.proto";
message LoginRequest
{
	optional string name = 1;
	optional string pwd = 2;
}

//登录结果枚举
enum enumLoginResult
{
	enumLoginResult_Success = 0;
	enumLoginResult_Fail = 1;
	enumLoginResult_NameFail = 2;
	enumLoginResult_PwdFail = 3;
	enumLoginResult_HaveLogin = 4;	//已经登录
};


message LoginResponse
{
	optional string gate_ip = 1;
	optional uint32 gate_port = 2;
	optional uint32 user_account = 3;		//账号ID
	optional enumLoginResult result = 4;	//登录结果
	optional uint32 chartype = 5;
};

//登录Gate
message LoginGame
{
	required uint32 user_account = 1;		//账号ID
};

//玩家初始化信息
message ClientInit
{
	optional CharInfo charinfo = 1;
	optional enumGetCharResult result = 2;
};

//客户端更新创建角色信息
message CreateChar
{
	optional CharInfo charinfo = 1;	  //创建角色
	optional uint32 mapid = 2;
	optional uint32 cityid = 3;
};

//离开服务器
message ClientExit
{
	optional uint32 charid = 1;
	optional uint64 user_account = 2;
};


//消息的类型
enum enumChatType
{
	Chat_User = 1;		//点对点聊天
	Chat_World = 2;		//世界聊天
	Chat_Sys = 3;		//系统消息
};

//聊天消息, 客户端发送给服务器，服务器再广播出去
message SendChat
{
	optional enumChatType type  = 1;
	optional uint32 charid = 2;
	optional string msg = 3;
};
//打招呼
message  SendGreet
{
	optional uint32 charid = 1; //角色id
	optional uint32 greetid = 2;  //招呼id
};
//公司急聘
message WordWindow
{
	optional enumWordmsgType type = 1;
	optional string content =2;
};
//切换服装
message ChangeCloth
{
	optional uint32 clothid = 1;
	optional uint32 charid = 2;
};
//评价
message Assess
{
	optional enumAssessType assesstype = 1;
	optional enumCharType  chartype = 2;
	optional uint32 id = 3;
	optional uint32 rank =4;
};
